cpp/src/join/mixed_join_semi.cu

```diff
@@ -160,22 +121,16 @@ std::unique_ptr<rmm::device_uvector<size_type>> mixed_join_semi(
   auto const preprocessed_probe =
     experimental::row::equality::preprocessed_table::create(probe, stream);
   auto const row_comparator =
-    cudf::experimental::row::equality::two_table_comparator{preprocessed_probe, preprocessed_build};
+    cudf::experimental::row::equality::two_table_comparator{preprocessed_build, preprocessed_probe};
```

Comment:
I need to dig into why but without this change the gtests start failing. Maybe @srinivasyadav18 can provide us a quick insight.

Reply:
Update: Shouldn't have changed the order here and instead needed to set `swap_tables=true` at mixed_join_kernels_semi.cu:+72. This is because we reset the hasher and equality functors for the hash_set swapping the `left` and `right` directions of the comparisons.

Reply:
Doing the swap here was just putting a patch on the underlying problem which was being caught in Java tests. I replicated the problem in C++ tests and have added the new test too so it can be caught early on.

Reply:
This is mainly due to a breaking change in https://github.com/NVIDIA/cuCollections/pull/479 where we always put query keys on the left-hand side for comparison.